android/libraries/rib-base/src/main/java/com/badoo/ribs/android/recyclerview/RecyclerViewHostView.kt

```diff
@@ -0,0 +1,35 @@
+package com.badoo.ribs.android.recyclerview
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.badoo.ribs.android.recyclerview.RecyclerViewHost.Dependency.LayoutManagerFactory
+import com.badoo.ribs.android.recyclerview.RecyclerViewHostView.Dependency
+import com.badoo.ribs.core.view.RibView
+import com.badoo.ribs.core.view.ViewFactory
+
+
+internal interface RecyclerViewHostView : RibView {
+
+    interface Factory : ViewFactory<Dependency, RecyclerViewHostView>
+
+    interface Dependency {
+        fun adapter(): Adapter<*>
+        fun layoutManagerFactory(): LayoutManagerFactory
+    }
+}
+
+internal class RecyclerViewHostViewImpl private constructor(
+    override val androidView: RecyclerView
+) : RecyclerViewHostView {
+
+    class Factory: RecyclerViewHostView.Factory {
+        override fun invoke(deps: Dependency): (ViewGroup) -> RecyclerViewHostView = {
+            RecyclerViewHostViewImpl(
+                androidView = RecyclerView(it.context).apply {
+                    adapter = deps.adapter()
```

Comment:
how can we modify recyclerView? can it be part of viewDependency? Because how we can solve problem when we need to add item decorator, item Animator or change snap?

Reply:
Good catch. I can lift it out to dependency similar to LayoutManager.